pkg/keccak/contrib/Modes/keccak1600.h

```diff
@@ -0,0 +1,169 @@
+  * @param  hash_instance  Pointer to the hash instance to be initialized.
+  */
+static inline void shake128_initialize(keccak1600hash_instance *hash_instance) {
+    keccak1600hash_initialize(hash_instance, 1344, 256, 0, 0x1F);
+}
+
+/** Function to initialize a SHAKE256 instance as specified in the FIPS 202 standard.
+  * @param  hash_instance  Pointer to the hash instance to be initialized.
+  */
+static inline void shake256_initialize(keccak1600hash_instance *hash_instance) {
+    keccak1600hash_initialize(hash_instance, 1088, 512, 0, 0x1F);
+}
+
+/** Function to initialize a Keccak-1600 instance for hashing with a security level of 128 bits.
+  * @param  hash_instance  Pointer to the hash instance to be initialized.
+  */
+static inline void sha3_128_initialize(keccak1600hash_instance *hash_instance) {
+    keccak1600hash_initialize(hash_instance, 1344, 256, 128, 0x06);
+}
+
+/** Function to initialize a SHA3-224 instance as specified in the FIPS 202 standard.
+  * @param  hash_instance  Pointer to the hash instance to be initialized.
+  */
+static inline void sha3_224_initialize(keccak1600hash_instance *hash_instance) {
+    keccak1600hash_initialize(hash_instance, 1152, 448, 224, 0x06);
+}
+
+/** Function to initialize a SHA3-256 instance as specified in the FIPS 202 standard.
+  * @param  hash_instance  Pointer to the hash instance to be initialized.
+  */
+static inline void sha3_256_initialize(keccak1600hash_instance *hash_instance) {
+    keccak1600hash_initialize(hash_instance, 1088, 512, 256, 0x06);
+}
+
+/** Function to initialize a SHA3-384 instance as specified in the FIPS 202 standard.
+  * @param  hash_instance  Pointer to the hash instance to be initialized.
+  */
+static inline void sha3_384_initialize(keccak1600hash_instance *hash_instance) {
+    keccak1600hash_initialize(hash_instance, 832, 768, 384, 0x06);
+}
+
+/** Function to initialize a SHA3-384 instance as specified in the FIPS 202 standard.
+  * @param  hash_instance  Pointer to the hash instance to be initialized.
+  */
+static inline void sha3_512_initialize(keccak1600hash_instance *hash_instance) {
+    keccak1600hash_initialize(hash_instance, 576, 1024, 512, 0x06);
+}
+
+/**
+  * Function to give input data to be absorbed.
+  * @param  hash_instance    Pointer to the hash instance initialized by keccak1600hash_initialize().
+  * @param  data        Pointer to the input data.
+  *                     When @a databitLen is not a multiple of 8, the last bits of data must be
+  *                     in the least significant bits of the last byte (little-endian convention).
+  * @param  databitLen  The number of input bits provided in the input data.
+  * @pre    In the previous call to keccak1600hash_update(), databitlen was a multiple of 8.
+  * @return SUCCESS if successful, FAIL otherwise.
+  */
+#define keccak1600hash_update(hash_instance, data, databitlen) Keccak_HashUpdate(hash_instance, data, databitlen)
```

Comment:
Required? If yes, why not an inline function as SHA-3 functions above?

Reply:
See above.